en/lessons/basics/functions.md

```diff
@@ -1,425 +1,83 @@
 ---
-version: 1.2.0
+version: 1.0.0
 title: Functions
 ---
 
-In Elixir and many functional languages, functions are first class citizens.
-We will learn about the types of functions in Elixir, what makes them different, and how to use them.
+Functions are at the heart of every Haskell program. Let us now see how to use them.
 
 {% include toc.html %}
 
 ## Anonymous Functions
 
-Just as the name implies, an anonymous function has no name.
-As we saw in the `Enum` lesson, these are frequently passed to other functions.
-To define an anonymous function in Elixir we need the `fn` and `end` keywords.
-Within these we can define any number of parameters and function bodies separated by `->`.
+Lambdas are one of the building blocks of Haskell, and their syntax is the following:
 
-Let's look at a basic example:
-
-```elixir
-iex> sum = fn (a, b) -> a + b end
-iex> sum.(2, 3)
-5
 ```
+   Structure of a lambda
 
-### The & Shorthand
-
-Using anonymous functions is such a common practice in Elixir there is shorthand for doing so:
-
-```elixir
-iex> sum = &(&1 + &2)
-iex> sum.(2, 3)
-5
-```
-
-As you probably already guessed, in the shorthand version our parameters are available to us as `&1`, `&2`, `&3`, and so on.
-
-## Pattern Matching
-
-Pattern matching isn't limited to just variables in Elixir, it can be applied to function signatures as we will see in this section.
-
-Elixir uses pattern matching to check through all possible match options and select the first matching option to run:
-
-```elixir
-iex> handle_result = fn
-...>   {:ok, result} -> IO.puts "Handling result..."
-...>   {:ok, _} -> IO.puts "This would be never run as previous will be matched beforehand."
-...>   {:error} -> IO.puts "An error has occurred!"
-...> end
-
-iex> some_result = 1
-1
-iex> handle_result.({:ok, some_result})
-Handling result...
-:ok
-iex> handle_result.({:error})
-An error has occurred!
+   \x -> x
+    │ │  └─ Body of the lambda
+    │ └─ Separator between head and body of the lambda
+    └─ Declaration of the input bindigs
 ```
 
-## Named Functions
-
-We can define functions with names so we can easily refer to them later.
-Named functions are defined within a module using the `def` keyword .
-We'll learn more about Modules in the next lessons, for now we'll focus on the named functions alone.
+And incidentally, we have also seen one of the most primordial functions : The identity function, or `id`.  
```

Comment:
`:` shouldn't have a space before it, and perhaps "you have just seen" would better highlight that you're referring to `\x -> x`. 

I'm also not sure about the usage of "we" here (and in general): I haven't needed to use it in `basics.md`, but if I were to use it, I'd use it to refer to the general Haskell userbase or the writers of the article (e.g. "We introduced").

Reply:
Ok!

Reply:
I think that "The" shouldn't be capitalised here.